contracts/on-chain-claim/src/config.rs

```diff
@@ -40,16 +71,47 @@ pub trait ConfigModule {
             return false;
         }
 
-        let address_info = address_info_mapper.get();
+        let address_info = self.get_address_info(address);
+        if address_info.total_epochs_claimed == 0 {
+            return false;
+        }
+
         let current_epoch = self.blockchain().get_block_epoch();
         let missed_epochs = self.get_missed_epochs(current_epoch, address_info.last_epoch_claimed);
 
         missed_epochs > 0 && missed_epochs <= MAX_REPAIR_GAP
     }
 
+    #[view(getCurrentSeason)]
+    fn get_current_season(&self) -> SeasonInfo {
+        let current_epoch = self.blockchain().get_block_epoch();
+        let seasons = self.seasons().get();
+
+        let mut current_season: SeasonInfo = SeasonInfo::new(1u16, 0u64);
+
+        for season in seasons.iter() {
+            if season.start_epoch <= current_epoch {
+                current_season = season;
+            }
+        }
+
+        current_season
+    }
+
     #[storage_mapper("address_info")]
```

Comment:
Add view here for backwards compatibility purposes.

Reply:
Updated views now take the legacy storage into account, no need to add the view here as well.